### PR TITLE
check response from parseCSSColor before downgrading it as it could be null

### PR DESCRIPTION
**Pull request by andrewharvey**

### CODE REVIEW

**Comment by lucaswoj** [comment]
This looks great! Thank you @andrewharvey.

I just have one small request before merging: `parseColor` was recently factored out of `StyleDeclaration` and currently all our [color parsing tests live in `style_declaration.test.js`](https://github.com/mapbox/mapbox-gl-js/blob/master/test/js/style/style_declaration.test.js#L46-L58). Could you either move all the tests to `parse_color.test.js` or move your test to `style_declaration.test.js`?

**Comment by andrewharvey** [comment]
@lucaswoj Oh I didn't see those tests in style_declaration.test.js, I've updated the commit to include my test in with the others.

**Comment by lucaswoj** [comment]
Looks great to me @andrewharvey! :bow: 

I'm going to give the rest of the team a few more hours to chime in, then merge this into master.

File: js/style/parse_color.js [hunk 1]
```diff
@@ -22,9 +22,14 @@ function parseColor(input) {
 
     // Color string
     } else if (typeof input === 'string') {
-        var output = colorDowngrade(parseCSSColor(input));
-        colorCache[input] = output;
-        return output;
+        var parsedColor = parseCSSColor(input);
+        if (parsedColor !== null) {
```
Comment (jfirebaugh) [inline js/style/parse_color.js]: Use a guard clause (`if (!parsedColor) { throw new Error('Invalid color ' + input); }`) and unindent the happy code path.

Comment (andrewharvey) [inline js/style/parse_color.js]: Thanks for the style advise. Agreed, much better that way. I've updated the commit.

**Comment by jfirebaugh** [comment]
:+1: Thank you!

**Comment by lucaswoj** [comment]
:ship: in 6a187345032e387bd473ceb0f65a44de87f6c294

**Pull request closed without merging**